### PR TITLE
1.19.4 update: Fabric update, Update Gradle, Relicense to LGPLv3

### PR DESCRIPTION
**Pull request by diademiemi**
This pull request resolves https://github.com/WildfireRomeo/WildfireFemaleGenderMod/issues/110

Includes patches from @Kichura in https://github.com/WildfireRomeo/WildfireFemaleGenderMod/issues/110#issuecomment-1487483914

Will need to be merged into a new fabric-1.19.4 branch, but it does not exist yet so I can't target it yet.

### CODE REVIEW

**Comment by Kichura** [comment]
> Hmm after that patch it doesn't seem to launch anymore, not sure what commit caused it

Seems to be a mixin failure that is inside the mixins directory itself as that is what it's saying, unless gradle changed something but that shouldn't happen on purpose.

**Comment by diademiemi** [comment]
Reverting [`fecc775` (#111)](https://github.com/WildfireRomeo/WildfireFemaleGenderMod/pull/111/commits/fecc77587a46335c93ea605f7859381036cb9f98) fixes the issue

**Comment by Kichura** [comment]
> Reverting [`fecc775` (#111)](https://github.com/WildfireRomeo/WildfireFemaleGenderMod/pull/111/commits/fecc77587a46335c93ea605f7859381036cb9f98) fixes the issue

does gradle 7.5.1 cause same problem?

**Comment by Kichura** [comment]
> Fabric itself doesn't use Gradle 7.6.1. I just pushed an update to 7.5.1 which does work correctly, Fabric also uses this version: [FabricMC/fabric@`1.19.4`/gradle/wrapper/gradle-wrapper.properties](https://github.com/FabricMC/fabric/blob/1.19.4/gradle/wrapper/gradle-wrapper.properties?rgh-link-date=2023-03-29T20%3A16%3A30Z)

will push a quick fix for jar itself then.

**Comment by diademiemi** [comment]
This PR includes a license change from GPLv3 to LGPLv3 as per Modrinths guidelines, maybe @Kichura can explain more about why this is needed? I could not find where on Modrinth it's stated, but I assume this is to allow it to be used in modpacks. 

Required signoffs for license change:
- [ ] @pupnewfster 
- [ ] @WildfireRomeo 
- [x] @Kichura 
- [ ] @3x1t-5tyl3 
Please comment below confirming you agree with this license change before we can proceed with this PR!

**Comment by diademiemi** [comment]
Also, can a branch fabric-1.19.4 be created? Or will we rename fabric-1.19.3 after this PR?

**Comment by Kichura** [comment]
> Also, can a branch fabric-1.19.4 be created? Or will we rename fabric-1.19.3 after this PR?

Yes the branch is creatable if wildfire wants to request branch creation by PRs.

**Comment by Scraft161** [comment]
GPL is not compatible due to how it requires the base game to be under a copyleft license, LGPL does not have this requirement.
while GPL is not compatible in a literal sense, this can be fixed by either using LGPL or adding a GPL exception, both are valid solutions.

I hold no code in this project so I don't hold any stake in this argument but I have not seen a case where this has been a problem yet, truth is that large companies don't really care about GPL violations but making sure projects like these have their bases covered is always a good thing.

What modrinth does in regards to project licensing shouldn't be too relevant in the big picture as they are just a distribution platform, but they hold the power to take anything down that they don't like (this is how any platform like theirs works).
I don't know if they've given a clear explanation as of why they want you to change the license, but in cases like these it is at least warranted even if it's a simple link to a post explaining why.

tl;dr
mod licensing is always a bit of a nothingburger as barely anyone really cares, but making sure you've got your basis covered even for a technicality is never a bad idea since copyright and code mix like oil and water anyways.

**Comment by diademiemi** [comment]
I see, makes sense, thanks both of you!

**Comment by 3x1t-5tyl3** [comment]
Good on my side. 👍

**Comment by pupnewfster** [comment]
I agree to relicense my contributions to Wildfire's Female Gender Mod under LGPLv3

**Comment by WildfireRomeo** [comment]
Thanks for the pull request, I will double check it to make sure it's good before merging.

**Comment by Taureon** [comment]
hey i tried to build the repository by diademiemi that this wants to commit from so i dont have to wait, but i get this error?
what do i do here? i dont use java and i have been told i should just run gradlew.bat and it should be done
image

**Comment by Kichura** [comment]
> hey i tried to build the repository by diademiemi that this wants to commit from so i dont have to wait, but i get this error? what do i do here? i dont use java and i have been told i should just run gradlew.bat and it should be done image

Your gradle is complaining that you don't have JDK 17 during compile attempt.

**Comment by Taureon** [comment]
> > hey i tried to build the repository by diademiemi that this wants to commit from so i dont have to wait, but i get this error? what do i do here? i dont use java and i have been told i should just run gradlew.bat and it should be done image
> 
> Your gradle is complaining that you don't have JDK 17 during compile attempt.

I have installed JDK 17 from [oracle themselves](https://www.oracle.com/java/technologies/downloads/#jdk17-windows) and it showed the exact same error (except the "build failed in Xms" being different ofc) after running gradlew.bat again
image

**Comment by Kichura** [comment]
> > > hey i tried to build the repository by diademiemi that this wants to commit from so i dont have to wait, but i get this error? what do i do here? i dont use java and i have been told i should just run gradlew.bat and it should be done image
> > 
> > 
> > Your gradle is complaining that you don't have JDK 17 during compile attempt.
> 
> I have installed JDK 17 from [oracle themselves](https://www.oracle.com/java/technologies/downloads/#jdk17-windows) and it showed the exact same error (except the "build failed in Xms" being different ofc) after running gradlew.bat again image

Please retry this with Liberica JDK 17 instead as oracle does not add the `JAVA_HOME` path to your machine by default.

**Comment by OliveTheFemboy** [comment]
I have just built the pull request on Linux. Everything went fine, but I don't know where the build was saved to. This is my first time using gradle, so I'm not sure if I've missed something or if I'm forgetting another step to make the final .jar.

Also, here's some terminal output in case I've missed a step.
image_2023-05-20_083154443

**Comment by Kichura** [comment]
> I have just built the pull request on Linux. Everything went fine, but I don't know where the build was saved to. This is my first time using gradle, so I'm not sure if I've missed something or if I'm forgetting another step to make the final .jar.
> 
> Also, here's some terminal output in case I've missed a step. image_2023-05-20_083154443

Final jars in gradle are always stored inside `WildFireFemaleGenderMod-X-Y/build/libs` and the actual mod must not be a dev nor sources jar.

**Comment by Taureon** [comment]
> Please retry this with Liberica JDK 17 instead as oracle does not add the `JAVA_HOME` path to your machine by default, in fact you should check your `java --version` information in case it isn't a fluke.

lmao i give up
image

**Comment by Kichura** [comment]
> > Please retry this with Liberica JDK 17 instead as oracle does not add the `JAVA_HOME` path to your machine by default, in fact you should check your `java --version` information in case it isn't a fluke.
> 
> lmao i give up image

I mean you could just do `gradlew.bat -Dorg.gradle.java.home="C:\Program Files\BellSoft\LibericaJDK-17" build`.

**Comment by Taureon** [comment]
> I mean you could just do `gradlew.bat -Dorg.gradle.java.home="C:\Program Files\BellSoft\LibericaJDK-17" build`.

image
image

**Comment by Kichura** [comment]
> > I mean you could just do `gradlew.bat -Dorg.gradle.java.home="C:\Program Files\BellSoft\LibericaJDK-17" build`.
> > i wonder which system variable gradle is looking at for doing this
> > image
> > image

Seems like your gradle isn't adapting to the JDK configurations properly, But there is a pre-built mod of this in the discord via #mod-discussion if you wish to obtain it this way instead.

**Comment by OliveTheFemboy** [comment]
> > I have just built the pull request on Linux. Everything went fine, but I don't know where the build was saved to. This is my first time using gradle, so I'm not sure if I've missed something or if I'm forgetting another step to make the final .jar.
> > Also, here's some terminal output in case I've missed a step. image_2023-05-20_083154443
> 
> Final jars in gradle are always stored inside `WildFireFemaleGenderMod-X-Y/build/libs` and the actual mod must not be a dev nor sources jar.

I firgured out my error and why nothing was created. I forgot to add "assemble" to the command, so nothing was actually outputted beyond an empty loom-cache folder.

**Comment by Kichura** [comment]
Update: This PR seems to be skipped over and may end up in closure if nothing else is coming through from what i am believing.

**Comment by WildfireRomeo** [comment]
I did unfortunately skip over this PR as I'm gonna be honest, I completely forgot about it.

I'll try to be better about remembering PR's if I do want to add them in the future, for this I'll probably just try to edit the files manually with the PR commits and just close it so there isn't a merging mess at this point.
